rclcpp/include/rclcpp/topic_statistics/subscription_topic_statistics.hpp

```diff
@@ -136,13 +135,10 @@ class SubscriptionTopicStatistics
           window_start_,
           window_end,
           collected_stats);
-        msgs.push_back(message);
+        publisher_->publish(message);
```

Comment:
Moving `publish` to send the message for each message types, this mitigates the message order inversion and losing message with QoS history keep_last setting reported in https://github.com/ros2/rclcpp/issues/2588.
This moves the `publisher_` under the lock contention, but i think that is also fine because `publisher_` is only constructed by ctor and destructed during tear_down.